### PR TITLE
check_sftp: fixed validation for private-key arg

### PR DESCRIPTION
**Pull request by wu3rstle**
corrected argument name to expect to prevent the error: `option --private-key not recognized`

### CODE REVIEW

**Comment by jherbel** [comment]
Hi, thanks for this PR. Are you actually using this argument? I am asking since it is not configurable via the GUI and it anyway does the same as --secret (which also looks like a bug to me).

**Comment by wu3rstle** [comment]
Hi, i am using this argument via **Integrate Nagios plugins** because we are connecting to an sftp server that only allows sftp authentication via ssh-key.

`check_sftp --host $HOSTADDRESS$ --user <SFTP_USER> --private-key ~/.ssh/id_rsa.openssh`

The two arguments are for different use cases:
* --secret: password for authentication
* --private-key: using ssh key for authentication

**Comment by jherbel** [comment]
Ok, this makes sense. However, then I think there is another bug in `check_sftp`. The code says:
```
elif opt in ["--secret"]:
	opt_pass = arg
elif opt in ["--private-key"]:
	opt_pass = arg
```
So `--secret` and `--private-key` effectively do the same thing (unintentionally). They are both passed on to `paramiko.client.SSHClient.connect` as `password`. I think `--private-key` should be set as `pkey` in `paramiko.client.SSHClient.connect`, right? The solution would be:
```
elif opt in ["--private-key"]:
	opt_key = arg
```
Could you please take a look and check I am correct? If I am, could you please include this in your fix and also briefly test it?

**Comment by wu3rstle** [comment]
i added the missing commit which was only local.

**Comment by wu3rstle** [comment]
Now the check on my local check_mk installation works as expected with the copied file.

Now i tell `paramiko.client.SSHClient.connect` to use the key as argument for `key_filename` to allow usage of the given ssh key path. 
I also adjusted the command's help message to make clear which key-format has to be used for the dependency.

File: active_checks/check_sftp [hunk 1]
```diff
@@ -21,7 +21,7 @@ OPTIONS:
   --host HOST            SFTP server address
   --user USER            Username for sftp login
   --secret SECRET        Secret/Password for sftp login
-  --private-key KEY      Private Key for sftp login
+  --private-key KEY      Path to rsa private Key file for sftp login
```
Comment (jherbel) [inline active_checks/check_sftp:24]: Lets call this private-key-file for clarity

Comment (jherbel) [inline active_checks/check_sftp:24]: Lets call this `--private-key-file PATH` for clarity

**Review by jherbel — CHANGES_REQUESTED** [review]
Hi, thanks for the changes, I would suggest to slightly rename the argument for clarity

File: active_checks/check_sftp (hunk 1 above)
Comment (wu3rstle) [reply active_checks/check_sftp]: makes sense to me.

I also had to adjust the whitespacing.

**Review by jherbel — APPROVED** [review]
Looks good, thanks, this will be merged shortly.

**Pull request closed without merging**